src/common/src/FileUpload/hooks.ts

```diff
@@ -11,42 +11,61 @@ export interface CustomFile extends File {
 export type CustomFileOrRejection = CustomFile & FileRejection;
 export type Files = CustomFileOrRejection[];
 
+let tmpId = 0;
+
 export const useFileUpload = (
 	getPostUrl: () => Promise<string>,
 	getHeaders: () => Promise<{ [key: string]: string }>,
 	httpMethod: 'POST' | 'PUT' = 'POST',
+	storedFiles: CustomFileOrRejection[] = [],
+	lastTmpId = 0,
 	onFileSuccess?: (file: CustomFile) => void,
 ) => {
-	const [files, setFiles] = React.useState<CustomFileOrRejection[]>([]);
+	const [files, setFiles] = React.useState<CustomFileOrRejection[]>(storedFiles);
 	const [stateXhr, setStateXhr] = React.useState<{ [key: string]: XMLHttpRequest }>({});
-	const [nextTmpId, setNextTmpId] = React.useState<number>(1);
+	tmpId = lastTmpId;
 
 	const handleOnDrop = React.useCallback(
 		(acceptedFiles: File[], fileRejections: FileRejection[]) => {
 			if (fileRejections.length) {
 				setFiles((previousFiles) => [...previousFiles, ...fileRejections] as Files);
 			}
 
-			const makeRequest = async (rawFile: File) => {
+			const makeRequest = async (rawFile: File, nextTmpId: number) => {
 				const postUrl = await getPostUrl();
 				const headers = await getHeaders();
-				const customFile: CustomFile = Object.assign(rawFile, { tmpId: nextTmpId });
-				const fileList = [...files, customFile] as Files;
-				setFiles(fileList);
+				const customFile: CustomFile = Object.assign(rawFile, { tmpId: nextTmpId, progress: 0 });
+				setFiles((previousFiles) => [...previousFiles, customFile] as Files);
 
 				const formData = new FormData();
 				formData.append('file', rawFile);
 				const xhr = new XMLHttpRequest();
 
 				xhr.upload.onprogress = (event) => {
-					customFile.progress = (event.loaded / event.total) * 100;
+					const percentage = parseInt(String((event.loaded / event.total) * 100), 10);
+					// Avoid file being re-rendered as removed from list prior to onreadystatechange having
+					// done its thing
+					if (percentage === 100) return;
+					setFiles(
+						(previousFiles) =>
+							[
+								...previousFiles.map((file) => {
+									if (file.tmpId === customFile.tmpId) {
```

Comment:
Kan er maar 1 concurrent file upload plaatsvinden?

Reply:
Multiple can be dropped at once. But each is uploaded with its own request and therefore XHR instance which gives progress updates.